### ISSUE TITLE
Static z-axis rotation

### ISSUE THREAD

**Issue by Kevin-Mattheus-Moerman** [JuliaGL/GLAbstraction.jl#96]
Sorry @SimonDanisch but this is a real stopper for me. The current view rotation settings are not useful and should frustrate any computational geometry person (see also: https://github.com/JuliaPlots/Makie.jl/issues/38, https://github.com/JuliaGL/GLAbstraction.jl/issues/76, https://github.com/JuliaGL/GLAbstraction.jl/pull/80, https://github.com/JuliaPlots/Makie.jl/issues/329). 
It is not just a matter of a opinion/preference, the current convention is actually not useful and is hindering use of the package. 

1. The current convention differs from all CAD packages and stuff like Blender and MeshLab. Currently Makie feels like somebody inverted the controls on your bike (or it is like everybody is used to riding a Lagrangian bike and currently we need to switch to a Eulerian bike). You might say this is just a preference but still it makes sense to match the standard convention which has been tried and tested and feels most natural to most people. Here is a GIF showing the MeshLab system: 
![Peek 2019-06-17 09-09_2](https://user-images.githubusercontent.com/8392709/59588693-1e7b9380-90e0-11e9-960e-496177c4136e.gif)

2. My main objection to the current convention is that not all views are obtainable i.e. the rotation system does not allow all types of orientations. For instance I challenge you to use the current system to make the cat look to the right so we can look at its left flank (while the cat is still facing forward). See my attempt in this GIF
![Peek 2019-06-17 09-09](https://user-images.githubusercontent.com/8392709/59588548-c47ace00-90df-11e9-878b-9fa343345651.gif)

3. When one continues to move the mouse in a single direction the direction of rotation should not change. Again if you look at the above GIF, if I move to cursor to the left the direction of rotation switches (perhaps something goes from 1 -> -1 or something after passing the middle?). The axis of rotation and angle should always be based on the cursor motion vector not location. 

In short please please please implement the standard CAD convention.

**Comment by SimonDanisch** [comment]
Yeah would be nice! It's very hard to get a good reference for camera code, since it's either hidden deep in a complex C++ library, closed source, or one just finds toy example. I even went through the effort to extract the camera code from Blender, but it was so complex I gave up^^

So this was definitely one of the occasions where I had 0 motivation to reinvent the wheel, but I couldn't find any good reference to implement this nicely.
At some point I was just like: ok I wasted way too much time on the camera code, this has to be good enough for now. 
Any help to improve this would be very welcome.

Btw, at some point I implemented the cube camera navigation from 3Ds max, which was pretty nice, but integrated rather awkwardly with the rest....

**Comment by Kevin-Mattheus-Moerman** [comment]
I did not realize this is difficult to do. I implemented a similar fix for MATLAB based rotation (https://www.gibboncode.org/img/gibbonViewControl.gif, left GIBBON, right MATLAB). Since I'm new to Julia I might not be much help but I can have a look if you like. Could you point me to the right code?

**Comment by SimonDanisch** [comment]
Well, it's probably not super hard, but starting from scratch without any mathematical background was definitely not easy, especially if one doesn't really know how the best camera should be ;) 
Here lives the 3D camera code:
https://github.com/JuliaPlots/AbstractPlotting.jl/blob/master/src/camera/camera3d.jl

**Comment by Kevin-Mattheus-Moerman** [comment]
Thanks. I'll have a look.

**Comment by Kevin-Mattheus-Moerman** [comment]
Thanks @SimonDanisch I've moved the discussion over to: https://github.com/JuliaPlots/AbstractPlotting.jl/issues/147
